### PR TITLE
IMP/Fix

### PR DESCRIPTION
**Pull request by vishal-onnet**
Improved the code and fixed the issue when first time record is being created then jsignature library was not properly initialized.

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@MeetSCS ping.

File: web_digital_sign/__manifest__.py [hunk 1]
```diff
@@ -8,10 +8,20 @@
     'license': 'AGPL-3',
     'complexity': 'easy',
     'depends': ['web'],
-    'description': ''' This module provides the functionality to store
-        digital signature image for a record.
```
Comment (ghost) [inline web_digital_sign/__manifest__.py:12]: Replace signature image with only signature.

File: web_digital_sign/static/description/index.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<section class="oe_container">
+    <div class="oe_row oe_spaced">
+        <div class="oe_span10">
+            <h2 class="oe_slogan">Web Digital Signature</h2>
+            <h3 class="oe_slogan">This module provides the functionality to store digital signature image for a record</h3>
```
Comment (ghost) [inline web_digital_sign/static/description/index.html]: signature image -> signature

File: web_digital_sign/static/src/js/digital_sign.js [hunk 3]
```diff
@@ -2,163 +2,120 @@ odoo.define('web_digital_sign.web_digital_sign', function(require) {
...
+            var sign_options = {'background-color': '#fff','height':'150','width':'550'};
+            this.$el.find("#signature").jSignature("init",sign_options);
+            this.$el.find("#signature").attr({"tabindex": "0",'height':"100"});
+            this.empty_sign = this.$el.find("#signature").jSignature("getData",'image');
+            this.$el.find('#sign_clean').click(this.on_clear_sign);
+            this.$el.find('.save_sign').click(this.on_save_sign);
+        },
+        on_clear_sign: function() {
+            var self = this;
+            $(this.$el[0]).find(".signature > canvas").remove();
```
Comment (ghost) [inline web_digital_sign/static/src/js/digital_sign.js]: $el itself is jquery object so we don't need to wrap it again. We can directly use this.$el.find(".signature")

File: web_digital_sign/static/src/xml/digital_sign.xml [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+   <t t-name="FieldSignature">
+        <div class="panel panel-default mt16 mb0 " id="drawsign">
+            <div class="panel-heading oe_edit_only">
+                <div class="pull-right">
+                    <a id="sign_clean" class="btn btn-xs oe_edit_only">Clear</a>
+                    <a class="oe_edit_only save_sign"></a>
+                </div>
+                <strong>Draw your signature</strong>
+            </div>
+            <div id="signature" class="signature panel-body"></div>
```
Comment (ghost) [inline web_digital_sign/static/src/xml/digital_sign.xml]: Name of id and class is same.

File: web_digital_sign/static/src/xml/digital_sign.xml [hunk 5]
```diff
@@ -1,15 +1,23 @@
...
         </div>
     </t>
     <t t-name="FieldBinaryImage-extend">
-        <img t-att-src='url' t-att-border="widget.readonly ? 0 : 1" t-att-name="widget.name" t-att-width="widget.node.attrs.img_width || widget.node.attrs.width" t-att-height="widget.node.attrs.img_height || widget.node.attrs.height" />
+        <img t-att-src='url'
+            t-att-border="widget.readonly ? 0 : 1"
+            t-att-name="widget.name"
+            t-att-width="widget.node.attrs.img_width || widget.node.attrs.width"
+            t-att-height="widget.node.attrs.img_height || widget.node.attrs.height"
```
Comment (ghost) [inline web_digital_sign/static/src/xml/digital_sign.xml:20]: Can you please add tabindex attribute on img element ?

File: web_digital_sign/views/users_view.xml [hunk 6]
```diff
@@ -21,7 +21,7 @@
         <field name="inherit_id" ref="base.view_users_form_simple_modif" />
         <field name="arch" type="xml">
             <xpath expr="//field[@name='signature']" position="after">
-                <!--<label for="signature_image" class="oe_edit_only"/>-->
+                <label for="signature_image" class="oe_edit_only"/>
```
Comment (ghost) [inline web_digital_sign/views/users_view.xml:23]: Can you please explain this changes ?

File: web_digital_sign/static/src/js/digital_sign.js [hunk 7]
```diff
@@ -2,163 +2,120 @@ odoo.define('web_digital_sign.web_digital_sign', function(require) {
...
     var FieldSignature = core.form_widget_registry.map.image.extend({
         template: 'FieldSignature',
+        placeholder: "/web/static/src/img/placeholder.png",
+        initialize_content: function() {
+            var self = this;
+            this.$el.find('> img').remove();
+            this.$el.find('.signature > canvas').remove();
+            var sign_options = {'background-color': '#fff','height':'150','width':'550'};
+            this.$el.find("#signature").jSignature("init",sign_options);
+            this.$el.find("#signature").attr({"tabindex": "0",'height':"100"});
```
Comment (ghost) [inline web_digital_sign/static/src/js/digital_sign.js]: Tabindex should be added in xml schema.

**Review by ghost — CHANGES_REQUESTED** [review]

File: web_digital_sign/__manifest__.py (hunk 1 above)
Comment (vishal-onnet) [reply web_digital_sign/__manifest__.py:12]: Changes are done.

File: web_digital_sign/static/description/index.html (hunk 2 above)
Comment (vishal-onnet) [reply web_digital_sign/static/description/index.html]: Changes are done.

File: web_digital_sign/static/src/js/digital_sign.js (hunk 3 above)
Comment (vishal-onnet) [reply web_digital_sign/static/src/js/digital_sign.js]: Done. I have made changes according to where $(this.$el[0]) is defined.

File: web_digital_sign/static/src/xml/digital_sign.xml (hunk 4 above)
Comment (vishal-onnet) [reply web_digital_sign/static/src/xml/digital_sign.xml]: Replaced the attribute 'id' with 'class'.

File: web_digital_sign/static/src/xml/digital_sign.xml (hunk 5 above)
Comment (vishal-onnet) [reply web_digital_sign/static/src/xml/digital_sign.xml:20]: Changes are done.

**Comment by JayVora-SerpentCS** [comment]
guys, are we not done yet?

**Comment by ghost** [comment]
@JayVora-SerpentCS Travis is failing due to unknown reason. Same Travis issue is faced in other PRs. Need your assistance. I am also looking into it.

**Pull request merged**